Spigot-API-Patches/0219-NamespacedKey-based-plugin-chunk-tickets.patch

```diff
@@ -0,0 +1,280 @@
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/TickingLevel.java
+@@ -0,0 +1,62 @@
++package io.papermc.paper;
++
++import org.bukkit.Chunk;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents what kind of ticking is done on the {@link Chunk}.
++ * The important values have dedicated cached instances, eg. {@link #ENTITY}.
++ * Read their own documentation for more information.
++ * <p>
++ * Ordering is based on the impact/importance of instances,
++ * so eg. {@link #ENTITY} is placed before {@link #BLOCK} when sorted.
++ * Traits defined by a specific level are inherited in the levels placed before it.
++ * <p>
++ * Instances of this class are not singletons even though there are some
++ * cached instances, eg. {@link #ENTITY}.
++ */
++public class TickingLevel implements Comparable<TickingLevel> {
++
++    /**
++     * Both blocks and entities get ticked in this level.
++     */
++    public static final TickingLevel ENTITY = new TickingLevel(31);
++
++    /**
++     * Only blocks get ticked in this level, entities do not.
++     */
++    public static final TickingLevel BLOCK = new TickingLevel(32);
++
++    /**
++     * Nothing gets ticked in this level, but the {@link Chunk} stays loaded.
++     */
++    public static final TickingLevel NONE = new TickingLevel(33);
++
++    final int ticketLevel; //not (yet) exposed directly to the API
++
++    TickingLevel(int ticketLevel) {
++        this.ticketLevel = ticketLevel;
++    }
++
++    @Override
++    public int compareTo(@NotNull TickingLevel other) {
++        return Integer.compare(ticketLevel, other.ticketLevel);
++    }
++
++    @Override
++    public boolean equals(Object object) {
++        return object instanceof TickingLevel && compareTo((TickingLevel) object) == 0;
++    }
++
++    @Override
++    public int hashCode() {
++        return ticketLevel;
++    }
++
++    @Override
++    public String toString() {
++        return "TickingLevel{" +
++                "level=" + ticketLevel +
++                '}';
++    }
++}
+diff --git a/src/main/java/org/bukkit/Chunk.java b/src/main/java/org/bukkit/Chunk.java
+index 11b691cf59b4a72493c91055fd8c46d7681b4527..92ea6aed465d28e8fb6437540f84672600cad76d 100644
+--- a/src/main/java/org/bukkit/Chunk.java
++++ b/src/main/java/org/bukkit/Chunk.java
+@@ -239,6 +239,73 @@ public interface Chunk {
+     @NotNull
+     Collection<Plugin> getPluginChunkTickets();
+ 
++    // Paper start
++    /**
++     * Adds a {@link NamespacedKey} based plugin chunk ticket
++     * with {@link io.papermc.paper.TickingLevel#ENTITY}.
++     *
++     * @param key the identifier of the ticket
++     * @return {@code true} if the ticket was added,
++     * {@code false} if the same ticket already exists
++     * @see #addKeyedPluginChunkTicket(NamespacedKey, io.papermc.paper.TickingLevel)
++     */
++    boolean addKeyedPluginChunkTicket(@NotNull NamespacedKey key);
```

Comment:
I was confused on why these methods were added to Chunk since it makes little sense, but I see spigot added some here, so i guess for matching.

But per note below, please move the impls to be default methods in the api patch instead.